base_layer/wallet/src/wallet.rs

```diff
@@ -508,6 +514,28 @@ where
         Ok(tx_id)
     }
 
+    pub fn sign_message(
+        &mut self,
+        secret: RistrettoSecretKey,
+        nonce: RistrettoSecretKey,
+        message: &str,
+    ) -> Result<SchnorrSignature<RistrettoPublicKey, RistrettoSecretKey>, SchnorrSignatureError> {
+        let challenge = Blake256::digest(message.as_bytes());
+        RistrettoSchnorr::sign_raw(&secret, nonce, &challenge)
+    }
```

Comment:
We should be using the new API for new code

```suggestion
    pub fn sign_message(
        &mut self,
        secret: RistrettoSecretKey,
     
        message: &str,
    ) -> Result<SchnorrSignature<RistrettoPublicKey, RistrettoSecretKey>, SchnorrSignatureError> {
        let challenge = Blake256::digest(message.as_bytes());
        RistrettoSchnorr::sign_message(&secret, &challenge)
    }
```

Reply:
The signature uses weak fiat-shamir, which is exactly what the new API aims to prevent :)